src/vm/database/clarity_db.rs

```diff
@@ -135,6 +137,10 @@ impl BurnStateDB for &dyn BurnStateDB {
     ) -> Option<BurnchainHeaderHash> {
         (*self).get_burn_header_hash(height, sortition_id)
     }
+
+    fn get_stacks_epoch(&self, height: u32) -> Option<StacksEpoch> {
+        (*self).get_stacks_epoch(height)
```

Comment:
I think Rust dereferences as many as necessary - so not sure if the `(*self)` is necessary.

Reply:
Not here -- it's `&dyn BurnStateDB`, so you have to dereference explicitly in order to avoid recursion.